src/Badge.js

```diff
@@ -5,12 +5,16 @@ class Badge extends React.Component {
         children: PropTypes.oneOfType([
             PropTypes.element,
             PropTypes.string
-        ]).isRequired,
+        ]),
         text: PropTypes.string.isRequired
     }
 
     render() {
         var { children } = this.props;
+
+        // No badge if no children
+        if(!React.Children.count(children)) return null;
```

Comment:
Hmm weird. Do you know if there's already a bug about that on react?
Seems weird the behavior from a stateless component is different in React and in a Shallow renderer.